src/Tests/Compile/CompileEnv.test.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {strict as assert} from 'assert';
+
+import {CompilerBase} from '../../Backend/Compiler';
+import {Toolchains} from '../../Backend/Toolchain';
+import {ToolchainInfo} from '../../Backend/Toolchain';
+import {DebianToolchain} from '../../Backend/ToolchainImpl/DebianToolchain';
+import {CompileEnv} from '../../Compile/CompileEnv';
+import {Logger} from '../../Utils/Logger';
+
+class MockCompiler extends CompilerBase {
+  // TODO: What toolchain is necessary as tests?
+  installedToolchain: DebianToolchain;
+  availableToolchain: DebianToolchain;
+  ts: Toolchains;
+
+  constructor() {
+    super();
+    this.installedToolchain =
+        new DebianToolchain(new ToolchainInfo('npm', 'package manager for Node.js'));
+    this.availableToolchain = new DebianToolchain(
+        new ToolchainInfo('nodejs', 'Node.js event-based server-side javascript engine'));
+    this.ts = new Toolchains();
+    this.ts.push(this.installedToolchain);
+    this.ts.push(this.availableToolchain);
+  }
+  toolchains(): Toolchains {
+    return this.ts;
+  }
+};
+
+suite('Compile', function() {
+  suite('CompileEnv', function() {
+    const K_CLEANUP: string = 'cleanup';
+    const logger = new Logger();
+    const compiler = new MockCompiler();
+
+    suite('#constructor()', function() {
+      test('is constructed with params', function() {
+        let env = new CompileEnv(logger, compiler);
+        assert.equal(env.installed, undefined);
+        assert.strictEqual(env.compiler, compiler);
+      });
+    });
+
+    suite('#listAvailable()', function() {
+      test('lists available toolchains', function() {
+        let env = new CompileEnv(logger, compiler);
+        let toolchains = env.listAvailable();
+        assert.strictEqual(toolchains, compiler.toolchains());
```

Comment:
I wonder what will happen if the order of element inside `toolchains` is same with that of `compiler.toolchains()`.
If different order would fail this assert, I think something like the following would be better.
`toolchains.foreach(toolchain => { assert.isTrue(compiler.toolchains().includes(toolchain); })`

Reply:
FYI, when I write this test, I just considered the result of diff references.

If your point is a kind of problem about deep comparing, we can use deepEqual.

Reply:
Um, I don't understand how can be different between `toolchains` and `compiler.toolchains()`. The `env.listAvailable()` function just returns the `compiler.toolchains()`.So they are exactly same.